src/base/bittorrent/torrentcreatorthread.cpp

```diff
@@ -86,13 +89,33 @@ void TorrentCreatorThread::run()
     emit updateProgress(0);
 
     try {
-        libt::file_storage fs;
+        const QString parentPath = Utils::Fs::branchPath(m_params.inputPath) + "/";
+
         // Adding files to the torrent
-        libt::add_files(fs, Utils::Fs::toNativePath(m_params.inputPath).toStdString(), fileFilter);
+        libt::file_storage fs;
+        if (QFileInfo(m_params.inputPath).isFile()) {
+            libt::add_files(fs, Utils::Fs::toNativePath(m_params.inputPath).toStdString(), fileFilter);
+        }
+        else {
+            // need to sort the file names by natural sort order
+            QHash<QString, boost::int64_t> fileSizeMap;
+            QDirIterator dirIter(m_params.inputPath, (QDir::Files | QDir::NoDotAndDotDot), QDirIterator::Subdirectories);
+            while (dirIter.hasNext()) {
+                dirIter.next();
+                const QString relFilePath = dirIter.filePath().mid(parentPath.length());
+                fileSizeMap[relFilePath] = dirIter.fileInfo().size();
+            }
+
+            QStringList fileNames = fileSizeMap.keys();
+            std::sort(fileNames.begin(), fileNames.end(), Utils::String::naturalLessThan<Qt::CaseInsensitive>);
+            for (const auto &i : fileNames)
```

Comment:
Also it should be `qAsConst(fileNames)` to prevent container detaching (don't forget to add `#include "base/global.h"`).

Reply:
`i` is confusing here. Better use "fileName".

Reply:
done.